cli/command/service/logs.go

```diff
@@ -212,43 +235,90 @@ type logWriter struct {
 	opts *logsOptions
 	f    *taskFormatter
 	w    io.Writer
+
+	// detailsCache saves a pre-sorted list of the details keys, so we don't
+	// have to recompute their order every time. this operates under the
+	// assumption that the details keys are always the same (they should be)
```

Comment:
Perhaps in a future implementation, the set of keys could change over time. I think it's better not to encode this assumption into the client, especially if it's an optimization. I'd recommend either dropping the optimization, or finding a way to implement it without making the assumption (invalidate the cache if the set of keys changes, or something like that).